File: Introductory Problems/Weird Algorithm/Weird Algorithm.py
def solve(n):
    print(n, end=" ")
    while n != 1:
        if n % 2 == 0:
            n //= 2
        else:
            n = n * 3 + 1
        print(n, end=" ")


if __name__ == "__main__":
    n = int(input())
    solve(n)

File: Introductory Problems/Weird Algorithm/Weird Algorithm.cpp
#include <iostream>

using namespace std;

void solve(long long int &n)
{
    cout << n << " ";
    while (n != 1)
    {
        if (n % 2 == 0)
        {
            n /= 2;
        }
        else
        {
            n = n * 3 + 1;
        }
        cout << n << " ";
    }
}

int main()
{
    long long int n;
    cin >> n;
    solve(n);
}
